src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -740,7 +740,7 @@ class DeriveColumnDependencyError(Exception):
 def parse_column_name(column: Expression, analyzer: "Analyzer") -> Optional[str]:
     if isinstance(column, Expression):
         if isinstance(column, Attribute):
-            return column.name
+            return analyzer.analyze(column)
```

Comment:
I'm a little confused. Here we use analyzer to get the original name, right? But the column expression is "copied" and expression id is cleaned up, then how do we get the original name?

Reply:
For instance, in the case below, for `df2.select(df1["a"])`, `df1["a"]` will be converted to column name "b" instead of "a".

```
    session.sql_simplifier_enabled = True
    df1 = session.create_dataframe([[1]], schema=["a"])
    df2 = df1.select(df1["a"].alias("b"))
    df3 = df2.select(df1["a"])
    df3.show()
```

Reply:
Added the above code as code comment.